smartype-generator/src/main/kotlin/com/mparticle/smartype/generator/StringHelpers.kt

```diff
@@ -26,6 +26,9 @@ class StringHelpers {
                     uppercaseName = "${uppercaseName}${upperWord}"
                 }
             }
+            if (allUppercaseString) {
+                uppercaseName = uppercaseName.toUpperCase()
```

Comment:
I believe this logic needs to also be repeated above where we are returning early on line 10, in case there was an enum without underscores, only the first char would be uppercased.

Reply:
Actually it's not necessary because that specifically checks if there are 0 words, aka the string is empty. So it's just returning the original argument (which may either be null or an empty string)